### ISSUE TITLE
DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS and UnsafePreferInlineScheduling Support

### ISSUE THREAD

**Issue by Rodrigo-Andrade** [dotnet/runtime#59157]
I have being using this with great results for more than a year now, any plans for making this more "official"?

**Comment by stephentoub** [comment]
> I have being using this with great results

Can you share more details?

**Comment by Rodrigo-Andrade** [comment]
> Can you share more details?

I talked a bit about that here:
https://github.com/dotnet/runtime/issues/44449

But long story short: since the change of how sockets queue work in linux, my app had some performance regressions.
After enabling both flags, app performance doubled by avoiding queuing work in the threadpool.

**Comment by halter73** [comment]
At least for `UnsafePreferInlineScheduling` in Kestrel, the concern is this could cause worse performance and possibly even deadlocks given blocking code. I assume there are similar concerns for `DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS`.

Kestrel defaults to `AllowSynchronousIO` being false, but that just prevents calls to `HttpContext.Request.Body.Read(...)` and not `HttpContext.Request.Body.ReadAsync(...).Result` and other similar patterns. Even blocking on something like a database transaction or an outgoing HTTP request from a request handler could be a big problem if Kestrel's Socket transport doesn't dispatch to the thread pool.

**Comment by Rodrigo-Andrade** [comment]
Yes, and i had this problem (deadlocks), that actually led to improvements as some code paths were silently blocking.
But i am not proposing it to be the default, just that it is a legitimate option, right now it's my understanding that it's just experimental and can be pulled at any moment.

**Comment by karelz** [comment]
Triage: Currently we do not have plans to productize it, but we want to do more investigation in 7.0 + collect feedback from users who use, so we will keep it open.

**Comment by ShreyasJejurkar** [comment]
Please consider making this option as supported instead of experiment if not on by default.
